File: 0873/873.cs
public class Solution {
    public int LengthOfLongestFibonacciSubseq(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(LengthOfLongestFibonacciSubseq(root.left), LengthOfLongestFibonacciSubseq(root.right));
    }
}

File: 0873/873.c
int lengthOfLongestFibonacciSubseq(struct TreeNode* root) {
    if (!root) return 0;
    int l = lengthOfLongestFibonacciSubseq(root->left);
    int r = lengthOfLongestFibonacciSubseq(root->right);
    return 1 + (l > r ? l : r);
}